Qualifier on alternative bean - duplicate element error

### Describe the bug

I have a bean with annotations:

`@ApplicationScoped
@MyQualifier("some-value")
public class MyBean { .. }`

In the tests I have an alternative bean for it with following annotations:

`@ApplicationScoped
@MyQualifier("some-value")
@Alternative
@Priority(5)
public class TestMyBean extends MyBean { .. }`


When running the tests, an IllegalArgumentException happens with:
_duplicate element @MyQualifier("some-value")_

When I remove the qualifier from the alternative it works as expected.

### Expected behavior

I would expect that the Qualifier annotation can be defined on the alternative bean as it was prior to Quarkus 2.3
(Likely, the change was introduced with https://github.com/quarkusio/quarkus/pull/19759)

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

openjdk version "11.0.10" 2021-01-19 LTS

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.4.1.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

The exception is this:
`java.lang.RuntimeException: java.lang.ExceptionInInitializerError
Caused by: java.lang.ExceptionInInitializerError
Caused by: java.lang.RuntimeException: Failed to start quarkus
Caused by: java.lang.RuntimeException: Failed to initialize Arc
Caused by: java.lang.IllegalArgumentException: duplicate element: @MyQualifier(value="some-value")
`

But it seems that the issue is already resolved with Quarkus 2.5.x
(I can reproduce it with 2.4.1.Final but no longer with 2.5.1.Final)

--- a/independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Beans.java
+++ b/independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Beans.java
@@ -959,7 +959,8 @@ public final class Beans {
                 List<StereotypeInfo> stereotypes) {
             Collection<AnnotationInstance> qualifierCollection = beanDeployment.extractQualifiers(annotation);
             for (AnnotationInstance qualifierAnnotation : qualifierCollection) {
-                if (beanDeployment.isInheritedQualifier(qualifierAnnotation.name())) {
+                if (beanDeployment.isInheritedQualifier(qualifierAnnotation.name())
+                        && !Annotations.contains(qualifiers, qualifierAnnotation.name())) {
                     qualifiers.add(qualifierAnnotation);
                 }
             }
@@ -986,9 +987,9 @@ public final class Beans {
                     annotation = normalizedNamedQualifier(name, annotation);
                 }
             }
+            // Qualifiers
             Collection<AnnotationInstance> qualifierCollection = beanDeployment.extractQualifiers(annotation);
             for (AnnotationInstance qualifierAnnotation : qualifierCollection) {
-                // Qualifiers
                 qualifiers.add(qualifierAnnotation);
             }
             // Treat the case when an additional bean defining annotation that is also a qualifier declares the default scope
@@ -1040,10 +1041,10 @@ public final class Beans {
             List<StereotypeInfo> stereotypes = new ArrayList<>();
             Collection<AnnotationInstance> annotations = beanDeployment.getAnnotations(beanClass);
 
-            processSuperClass(beanClass, beanDeployment, qualifiers, stereotypes);
             for (AnnotationInstance annotation : annotations) {
                 processAnnotation(annotation, beanClass, beanDeployment, qualifiers, stereotypes, scopes);
             }
+            processSuperClass(beanClass, beanDeployment, qualifiers, stereotypes);
 
             if (scopes.size() > 1) {
                 throw multipleScopesFound("Bean class " + beanClass, scopes);

--- a/independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/inheritance/QualifiersInheritanceTest.java
+++ b/independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/inheritance/QualifiersInheritanceTest.java
@@ -1,0 +1,86 @@
+package io.quarkus.arc.test.inheritance;
+
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.ArcContainer;
+import io.quarkus.arc.test.ArcTestContainer;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.util.AnnotationLiteral;
+import javax.inject.Qualifier;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+public class QualifiersInheritanceTest {
+
+    @RegisterExtension
+    public ArcTestContainer container = ArcTestContainer.builder().removeUnusedBeans(false)
+            .beanClasses(SuperBean.class, Alpha.class, Bravo.class, InheritedQualifier.class, NonInheritedQualifier.class)
+            .build();
+
+    @SuppressWarnings("serial")
+    @Test
+    public void testInheritance() {
+        ArcContainer container = Arc.container();
+        // Bravo is not eligible because it has @InheritedQualifier("bravo")
+        assertTrue(container.select(SuperBean.class, new InheritedQualifier.Literal("super")).isResolvable());
+        // @NonInheritedQualifier is not inherited
+        assertFalse(container.select(SuperBean.class, new AnnotationLiteral<NonInheritedQualifier>() {
+        }).isResolvable());
+    }
+
+    @InheritedQualifier("super")
+    @NonInheritedQualifier
+    static class SuperBean {
+
+        public void ping() {
+        }
+    }
+
+    // -> inherited @InheritedQualifier("super")
+    @ApplicationScoped
+    static class Alpha extends SuperBean {
+
+    }
+
+    @InheritedQualifier("bravo")
+    @ApplicationScoped
+    static class Bravo extends SuperBean {
+
+    }
+
+    @Inherited
+    @Qualifier
+    @Retention(RUNTIME)
+    @interface InheritedQualifier {
+
+        String value();
+
+        @SuppressWarnings("serial")
+        static class Literal extends AnnotationLiteral<InheritedQualifier> implements InheritedQualifier {
+
+            private final String value;
+
+            public Literal(String value) {
+                this.value = value;
+            }
+
+            @Override
+            public String value() {
+                return value;
+            }
+
+        }
+
+    }
+
+    @Qualifier
+    @Retention(RUNTIME)
+    @interface NonInheritedQualifier {
+
+    }
+}

